rimage/lazy_encoded.go

```diff
@@ -31,7 +34,8 @@ type LazyEncodedImage struct {
 // from it. This is helpful for zero copy scenarios. If a width or height of the image is unknown,
 // pass 0 or -1; when done a decode will happen on Bounds. In the future this can probably go
 // away with reading all metadata from the header of the image bytes.
-// NOTE: Usage of an image that would fail to decode causes a lazy panic.
+// NOTE: It is recommended to call one of the Decode* methods and check for errors before using
+// image.Image methods (Bounds, ColorModel, or At) to avoid panics.
```

Comment:
heads up @bhaney - this should not change the current flow of image - but you will now be able to check if the image has been successfully Decoded from bytes into something that has bounds.